### PR TITLE
feat: ENT-4554 Added tableau api and iframe

### PR DESCRIPTION
**Pull request by irfanuddinahmad**
**Description**: This PR removes the tableau-react-embed package which had become un-supported and incorporates the latest tableau JS API script directly in our application.

**JIRA**: [ENT-4554](https://openedx.atlassian.net/browse/ENT-4554)

# For all changes

- [x] Ensure adequate tests are in place (or reviewed existing tests cover changes)

# Only if submitting a visual change

- [ ] Ensure to attach screenshots
- [ ] Ensure to have UX team confirm screenshots

### CODE REVIEW

File: src/components/analytics/AnalyticsCharts.jsx [hunk 1]
```diff
@@ -1,8 +1,7 @@
-import React, { useState, useEffect } from 'react';
+import React, { useState, useRef, useEffect } from 'react';
 import PropTypes from 'prop-types';
 import { logError } from '@edx/frontend-platform/logging';
-
-import TableauReport from 'tableau-react-embed';
+import url from 'url';
```
Comment (muhammad-ammar) [inline src/components/analytics/AnalyticsCharts.jsx:4]: what is the usage of this?

File: src/components/analytics/AnalyticsCharts.jsx [hunk 2]
```diff
@@ -11,33 +10,51 @@ import AnalyticsApiService from './data/service';
 
 // eslint-disable-next-line no-unused-vars
 export default function AnalyticsCharts(enterpriseId) {
-  const [token, setToken] = useState(null);
+  const [tokenUsedOnce, setTokenUsedOnce] = useState(false);
   const [isLoading, setIsLoading] = useState(false);
   const [error, setError] = useState(null);
-  // Fetch token
+  const ref = useRef(document.getElementById('tableau_frame'));
```
Comment (zamanafzal) [inline src/components/analytics/AnalyticsCharts.jsx:16]: I think this is not the right way of using this hook with `document.getElementById`. We can do it like this

```
  const tableauRef = useRef(null);
  <div id="tableau_frame" ref={tableauRef} />
```

**Review by zamanafzal — APPROVED** [review]

File: src/components/analytics/AnalyticsCharts.jsx (hunk 1 above)
Comment (irfanuddinahmad) [reply src/components/analytics/AnalyticsCharts.jsx:4]: To parse the url in getUrl

File: src/components/analytics/AnalyticsCharts.jsx (hunk 2 above)
Comment (irfanuddinahmad) [reply src/components/analytics/AnalyticsCharts.jsx]: done

**Review by muhammad-ammar — APPROVED** [review]
LGTM

**Pull request merged**